### PR TITLE
MAINT: update test standards

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Add python 3.11 testing, update nep29.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

via GA

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by jklenzing** [comment]
Link checks to wiley pubs are broken.  Need either an alternate record or we ignore this in checks

**Comment by JonathonMSmith** [comment]
Is this the link to the Huba et. al. paper, I don't see any failing link checks, so I guess this has been fixed?

**Comment by jklenzing** [comment]
> Is this the link to the Huba et. al. paper, I don't see any failing link checks, so I guess this has been fixed?

I've removed the links to AGU papers since they error on the link check. The full DOI should still be there. 

This only occurs in the rst files, links should still be in the readme

**Comment by jklenzing** [comment]
pinging @JonathonMSmith  for a review

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -3,20 +3,23 @@
...
 
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.9", "3.10"]
+        python-version: ["3.9", "3.10", "3.11"]
         numpy_ver: ["latest"]
         os: ["ubuntu-latest", "macos-latest", "windows-latest"]
         include:
           - python-version: "3.8"
```
Comment (jklenzing) [inline .github/workflows/main.yml]:
```suggestion
          - python-version: "3.9"
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -3,20 +3,23 @@
...
+  schedule:
+    - cron: "0 3 * * 1"  # Runs 03:00 UT on Mondays 
 jobs:
   build:
 
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.9", "3.10"]
+        python-version: ["3.9", "3.10", "3.11"]
```
Comment (jklenzing) [inline .github/workflows/main.yml]:
```suggestion
        python-version: ["3.10", "3.11"]
```

File: .readthedocs.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+
+# Required
+version: 2
+
+# Set the version of Python and other tools you might need
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "3.10"
```
Comment (jklenzing) [inline .readthedocs.yaml]:
```suggestion
    python: "3.11"
```

File: setup.cfg [hunk 4]
```diff
@@ -15,9 +15,10 @@ classifiers =
   Intended Audience :: Science/Research
   License :: OSI Approved :: BSD License
   Natural Language :: English
-  Programming Language :: Python :: 3.7
   Programming Language :: Python :: 3.8
```
Comment (jklenzing) [inline setup.cfg]:
```suggestion
```

**Review by jklenzing — COMMENTED** [review]
update for latest NEP29 standards

**Review by JonathonMSmith — APPROVED** [review]

**Pull request merged**